File: Part1/Laba_2/src/main/java/RunF1.java
public class RunF1 implements Runnable {
    private int[] B;
    private int[] C;
    private int[][] MA;
    private int[][] ME;
    private int[][] MD;
    Thread t;
    int n;

    public RunF1(int n) {
        t = new Thread(this, "F1");
        this.n = n;
        System.out.println("F1 started");
        t.start();
    }

    @Override
    public void run() {
        B = new int[n];
        C = new int[n];
        MA = new int[n][n];
        ME = new int[n][n];
        Data.generateVector(B);
        Data.generateVector(C);
        Data.generateMatrix(MA);
        Data.generateMatrix(ME);
        MD = Data.f1(B, C, MA, ME);
        System.out.println("F1 finished");
    }

    public void printResult() {
        System.out.println("\nResult of F1: MD = (B*C)*(MA*ME):\n");
        Data.printMatrix(MD);
    }
}

File: Part1/Laba_3/Laba_3/RunF1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Threading;

namespace Laba_3
{
    class RunF1
    {
        public Thread thread;
        int n;
        private int[] B;
        private int[] C;
        private int[][] MA;
        private int[][] ME;
        private int[][] MD;

        public RunF1(int n)
        {
            thread = new Thread(this.Run);
            this.n = n;
            Console.WriteLine("F1 started");
            thread.Start();
        }

        void Run()
        {
            B = new int[n];
            C = new int[n];
            MA = new int[n][];
            ME = new int[n][];
            Data.GenerateVector(B);
            Data.GenerateVector(C);
            Data.GenerateMatrix(MA);
            Data.GenerateMatrix(ME);
            MD = Data.F1(B, C, MA, ME);
            Console.WriteLine("F1 finished");
        }

        public void PrintResult()
        {
            Console.WriteLine("\nResult of F1: MD = (B*C)*(MA*ME):\n");
            Data.PrintMatrix(MD);
        }
    }
 }
